### ISSUE TITLE
Enhancement: PWA / mobile web app UI

### ISSUE THREAD

**Issue by andypiper** [ckolderup/postmarks#62]
This would be a great site to make available as a PWA for phones etc. Probably related to any other UI work that might be planned. Interested to help around these areas if I can, this is as much a note-to-self to look into as anything else.

**Comment by pburke** [comment]
One of the PWA features I'm most excited about is the [Web Share API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Share_API).

Right now it's a bit of a hurdle to add a bookmark to Postmarks from Firefox Mobile, and presumably other mobile browsers. It would be really nice to be able the use the native share prompt and select Postmarks (installed PWA) as the share target, which would populate the new bookmark form.

**Comment by ckolderup** [comment]
that would be pretty great! I'd be interested in doing this for sure; I think the UI refresh I've discussed with a couple frontend-leaning friends would certainly have mobile in mind more than I did when hacking out the UI we have now. So other than UI improvements this would involve:

* setting up the appropriate manifest file
* making sure we have appropriately sized images for setting the icon on various platforms
* setting up as a "web share target"

anything else that PWAs are capable of that would be interesting? I feel like a lot of the offline/caching stuff is kind of moot since so much of the site is about going out to other parts of the internet-- I think we're out of scope when we start talking about Instapaper-style offline reading.

**Comment by pburke** [comment]
Agreed about offline stuff. I can't think of any other PWA features that make sense right now.

Happy to pitch in on any part of the PWA setup or frontend work, just let me know.
